src/models/Particle.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Particle.cpp
+ *
+ *  Created on: 23.02.2010
+ *      Author: eckhardw
+ */
+
+#include "Particle.h"
+
+#include "utils/ArrayUtils.h"
+#include <iostream>
+
+Particle::Particle(int type_arg)
+{
+    type = type_arg;
+    std::cout << "Particle generated!" << std::endl;
+    f = { 0., 0., 0. };
+    old_f = { 0., 0., 0. };
+    m = 3;
+}
+
+Particle::Particle(const Particle& other)
+{
+    x = other.x;
+    v = other.v;
+    f = other.f;
+    old_f = other.old_f;
+    m = other.m;
+    type = other.type;
+    /* std::cout << "Particle generated by copy!" << std::endl; */
+}
+
+// Todo: maybe use initializater list instead of copy?
+Particle::Particle(
+    std::array<double, 3> x_arg, std::array<double, 3> v_arg, double m_arg, int type_arg)
+{
+    x = x_arg;
+    v = v_arg;
+    m = m_arg;
+    type = type_arg;
+    f = { 0., 0., 0. };
+    old_f = { 0., 0., 0. };
+    /* std::cout << "Particle generated!" << std::endl; */
+}
+
+Particle::~Particle()
+{
+    /* std::cout << "Particle destructed!" << std::endl; */
+}
+
+const std::array<double, 3>& Particle::getX() const
+{
+    return x;
+}
+
+const std::array<double, 3>& Particle::getV() const
+{
+    return v;
+}
+
+const std::array<double, 3>& Particle::getF() const
+{
+    return f;
+}
+
+const std::array<double, 3>& Particle::getOldF() const
+{
+    return old_f;
+}
+
+void Particle::setF(std::array<double, 3> f_new)
```

Comment:
just a minor recommendation here: I would make the argument as `const reference` as you don't plan to change the `f_new` anyway